packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts

```diff
@@ -149,7 +148,7 @@ export abstract class HandlebarsNodeVisitors extends Parser {
         break;
 
       default:
-        throw new Error(`Using a Handlebars comment when in the \`${tokenizer.state}\` state is not supported: "${comment.value}" on line ${loc.start.line}:${loc.start.column}`);
+        throw new SyntaxError(`Using a Handlebars comment when in the \`${tokenizer.state}\` state is not supported: "${comment.value}" on line ${loc.start.line}:${loc.start.column}`, rawComment.loc);
```

Comment:
Do you think we should still include the loc info inside the message now that we thread loc through the rest of the system?